torchgeo/datasets/flair2.py

```diff
@@ -0,0 +1,762 @@
+
+        return (
+            slice(x - eigth_size, x + eigth_size),
+            slice(y - eigth_size, y + eigth_size),
+        )
+
+    def _load_files(self) -> list[dict[str, Collection[str]]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image, masks
+        """
+        images = sorted(
+            glob.glob(
+                os.path.join(
+                    self.root,
+                    self.dir_names[self.split]['images'],
+                    '**',
+                    self.globs['images'],
+                ),
+                recursive=True,
+            )
+        )
+
+        sentinels_data = sorted(
+            glob.glob(
+                os.path.join(
+                    self.root,
+                    self.dir_names[self.split]['sentinels'],
+                    '**',
+                    self.globs['sentinels'].format('_data'),
+                ),
+                recursive=True,
+            )
+        )
+        sentinels_mask = sorted(
+            glob.glob(
+                os.path.join(
+                    self.root,
+                    self.dir_names[self.split]['sentinels'],
+                    '**',
+                    self.globs['sentinels'].format('_masks'),
+                ),
+                recursive=True,
+            )
+        )
+        sentinels = [
+            {'data': data, 'snow_cloud_mask': mask}
+            for data, mask in zip(sentinels_data, sentinels_mask)
+        ]
+
+        masks = sorted(
+            glob.glob(
+                os.path.join(
+                    self.root,
+                    self.dir_names[self.split]['masks'],
+                    '**',
+                    self.globs['masks'],
+                ),
+                recursive=True,
+            )
+        )
+
+        # One sentinel image might contain multiple aerial images, thus we need to match them
+        # without assuming a 1:1 mapping
+        sentinel_lookup = {'/'.join(s['data'].split('/')[-4:-2]): s for s in sentinels}
+        files = [
+            dict(
+                image=image,
+                sentinel=sentinel_lookup['/'.join(image.split('/')[-4:-2])],
+                mask=mask,
+            )
+            for image, mask in zip(images, masks)
+        ]
+
+        return files
+
+    def _load_image(self, path: Path) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            Tensor: the loaded image
+        """
+        with rasterio.open(path) as f:
+            array: np.typing.NDArray[np.int_] = f.read()
+            tensor = torch.from_numpy(array).float()
+            if 'B05' in self.bands:
+                # Height channel will always be the last dimension
+                tensor[-1] = torch.div(tensor[-1], 5)
+
+        return tensor
```

Comment:
Should the bands perhaps be extracted here based on `self.bands`? E.g., something like
```
    def _load_image(self, path: Path) -> Tensor:
        """Load a single image.
        Args:
            path: path to the image
        Returns:
            Tensor: the loaded image
        """
        with rasterio.open(path) as f:
            array: np.typing.NDArray[np.int_] = f.read()
            tensor = torch.from_numpy(array).float()
            if 'B05' in self.bands:
                # Height channel will always be the last dimension
                tensor[-1] = torch.div(tensor[-1], 5)

        # Extract the bands to be used. E.g., self.bands=("B01", "B02", "B03") will extract the RGB bands.
        tensor = tensor[[int(band[-2:]) - 1 for band in self.bands]]

        return tensor
``` 
Then when a user has defined n bands in `self.bands`, the returned sample will only contain those bands, instead of all five. Perhaps `self.bands` should also be renamed to `self.aerial_bands`, and a `self.sentinel_bands` should be added(?)